### ISSUE TITLE
Security update fails and Ubuntu APT ignores the APT link

### ISSUE THREAD

**Issue by ckhatton** [phusion/passenger#2498]
## Issue report

**Are you sure this is a bug in Passenger?**

Oh, yes.

**Please try with the newest version of Passenger to avoid issues that have already been fixed**

It's version 6.0.18.

**Question 1: What is the problem?**

- What is the expected behaviour?
  - It can carry out a security update.
  - Ubuntu can read the APT link

- What is the actual behaviour?
  - It fails to carry out a security update.
  - Ubuntu can't read the APT link

- How can we reproduce it? 
  - Check the Apache2 server log.
  - Run `sudo apt-get update`.

**Question 2: Passenger version and integration mode**:

- Standalone 6.0.18/Apache APT

**Question 3: OS or Linux distro, platform (including version)**:

  System:
    OS: Linux 5.4 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (4) x64 DO-Regular
    Container: Yes
    Shell: 3.6.1 - /usr/bin/fish

**Question 4: Passenger installation method**:

- [ ] RubyGems + Gemfile
- [ ] RubyGems, no Gemfile
- [x] Phusion APT repo
- [ ] Phusion YUM repo
- [ ] OS X Homebrew
- [ ] source tarball
- [ ] Other, please specify:

**Question 5: Your app's programming language (including any version managers) and framework (including versions)**:

N/A

**Question 6: Are you using a PaaS and/or containerization? If so which one?**

Nope.

**Question 7: Anything else about your setup that we should know?**

So when I restart Apache2, Passenger tries to carry out a security update and fails.

```bash
> sudo tail /var/log/apache2/error.log
[Tue Sep 12 13:06:37.065184 2023] [:notice] [pid 1612976:tid 140493649165376] FastCGI: process manager initialized (pid 1612976)
[ N 2023-09-12 13:06:37.1065 1612981/T1 age/Wat/WatchdogMain.cpp:1377 ]: Starting Passenger watchdog...
[ N 2023-09-12 13:06:37.1300 1612984/T1 age/Cor/CoreMain.cpp:1340 ]: Starting Passenger core...
[ N 2023-09-12 13:06:37.1302 1612984/T1 age/Cor/CoreMain.cpp:256 ]: Passenger core running in multi-application mode.
[ N 2023-09-12 13:06:37.1413 1612984/T1 age/Cor/CoreMain.cpp:1015 ]: Passenger core online, PID 1612984
[Tue Sep 12 13:06:37.144817 2023] [mpm_event:notice] [pid 1612974:tid 140493649165376] AH00489: Apache/2.4.41 (Ubuntu) mod_fastcgi/mod_fastcgi-SNAP-0910052141 mod_fcgid/2.3.9 OpenSSL/1.1.1f Phusion_Passenger/6.0.18 configured -- resuming normal operations
[Tue Sep 12 13:06:37.145052 2023] [core:notice] [pid 1612974:tid 140493649165376] AH00094: Command line: '/usr/sbin/apache2'
[ N 2023-09-12 13:06:37.1574 1612950/T1 age/Cor/TelemetryCollector.h:531 ]: Message from Phusion: End time can not be before or equal to begin time
[ N 2023-09-12 13:06:37.2095 1612950/T1 age/Cor/CoreMain.cpp:1325 ]: Passenger core shutdown finished
[ E 2023-09-12 13:06:39.1645 1612984/T7 age/Cor/SecurityUpdateChecker.h:506 ]: Security update check failed: Couldn't connect to server for https://securitycheck.phusionpassenger.com/v1/check.json (if this error persists check yourconnection security or try upgrading Passenger) (next check in 24 hours)
```

I look to carry out a manual upgrade and it complains the APT link is not correct for the system.

```bash
> sudo apt-get update
Hit:1 http://ppa.launchpad.net/ondrej/php/ubuntu focal InRelease
Hit:2 https://download.docker.com/linux/ubuntu focal InRelease
Hit:3 https://repos.insights.digitalocean.com/apt/do-agent main InRelease
Hit:4 http://mirrors.digitalocean.com/ubuntu focal InRelease
Ign:5 https://oss-binaries.phusionpassenger.com/apt/passenger focal InRelease
Hit:6 https://repos-droplet.digitalocean.com/apt/droplet-agent main InRelease
Hit:7 https://oss-binaries.phusionpassenger.com/apt/passenger focal Release
Hit:8 http://mirrors.digitalocean.com/ubuntu focal-updates InRelease
Hit:9 http://mirrors.digitalocean.com/ubuntu focal-backports InRelease
Hit:10 http://mirrors.digitalocean.com/ubuntu focal-security InRelease
Get:11 https://esm.ubuntu.com/infra/ubuntu focal-infra-security InRelease [7,453 B]
Get:12 https://esm.ubuntu.com/infra/ubuntu focal-infra-updates InRelease [7,452 B]
Fetched 14.9 kB in 2s (7,890 B/s)
Reading package lists... Done
N: Skipping acquire of configured file 'main/binary-i386/Packages' as repository 'https://oss-binaries.phusionpassenger.com/apt/passenger focal InRelease' doesn't support architecture 'i386'
```

So I add `[arch=amd64]` to `/etc/apt/sources.list.d/passenger.list`:

```sh
deb [arch=amd64] https://oss-binaries.phusionpassenger.com/apt/passenger focal main
```

And `apt` still ignores it:

```bash
> sudo apt-get update
Hit:1 http://ppa.launchpad.net/ondrej/php/ubuntu focal InRelease
Hit:2 https://download.docker.com/linux/ubuntu focal InRelease
Hit:3 https://repos.insights.digitalocean.com/apt/do-agent main InRelease
Hit:4 https://esm.ubuntu.com/infra/ubuntu focal-infra-security InRelease
Hit:5 https://esm.ubuntu.com/infra/ubuntu focal-infra-updates InRelease
Hit:6 https://repos-droplet.digitalocean.com/apt/droplet-agent main InRelease
Ign:7 https://oss-binaries.phusionpassenger.com/apt/passenger focal InRelease
Hit:8 http://mirrors.digitalocean.com/ubuntu focal InRelease
Hit:9 https://oss-binaries.phusionpassenger.com/apt/passenger focal Release
Hit:10 http://mirrors.digitalocean.com/ubuntu focal-updates InRelease
Hit:11 http://mirrors.digitalocean.com/ubuntu focal-backports InRelease
Hit:12 http://mirrors.digitalocean.com/ubuntu focal-security InRelease
Reading package lists... Done
```

**Comment by CamJN** [comment]
@FooBarWidget might your changes have affected this?

**Comment by FooBarWidget** [comment]
Hi, from your report I see different issues:

- The security update checker failing to query the server for update information. There was a period yesterday when the server was down for maintenance. It's currently up again. Do you still get the error now? 

- APT not showing any updates. This is normal because you are already on the latest version 6.0.18. I see no error messages by APT.

- The i386 warning is harmless. It's a warning and does not indicate any errors.

In short: everything your logs show indicate that the only problem is that the update information server was down yesterday, everything else is fine. I'm therefore closing this issue. If you have more questions or comments or disagree with this assessment, feel free to post.
